code/controllers/subsystem/mapping.dm

```diff
@@ -252,3 +256,96 @@ SUBSYSTEM_DEF(mapping)
 
 		shelter_templates[S.shelter_id] = S
 		map_templates[S.shelter_id] = S
+
+/*
+	Handles reserving an area of open space for you do do whatever with
+	Releases and clears out that area when Destroy()'d
+
+	Usage:
+
+	var/datum/area_reservation/R = SSmapping.RequestArea(width, height)    //assume this line sleeps
+
+	if(!R) return    //unable to reserve space
+
+	var/turf/bottom_right = R.bottom_right
+	var/turf/top_left = R.top_left
+
+	//do stuff with the area
+
+	qdel(R)
+*/
+
+/datum/controller/subsystem/mapping/proc/SetupReservedAreas()
+	// TODO: See if SHUTTLE_TRANSIT_BORDER is really necessary
+	var/turf/A = get_turf(locate(SHUTTLE_TRANSIT_BORDER, SHUTTLE_TRANSIT_BORDER, ZLEVEL_TRANSIT))
+	var/turf/B = get_turf(locate(world.maxx - SHUTTLE_TRANSIT_BORDER, world.maxy - SHUTTLE_TRANSIT_BORDER, ZLEVEL_TRANSIT))
+	unused_turfs = block(A, B)
+	var/datum/area_reservation/R = new
+	R.reserved_turfs = unused_turfs
+	qdel(R)	//this will prep the zone
+
+/datum/controller/subsystem/mapping/proc/RequestArea(width, height)
+	var/datum/area_reservation/R = new
+	if(!R.Reserve(width, height, unused_turfs, TRUE))
+		qdel(R)
+		return null
```

Comment:
just `return` will already return `null`, unless you really want it explicit for readability..?